scripts/world_bank/worldbank.py

```diff
@@ -0,0 +1,393 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Fetches, cleans, outputs TMCFs and CSVs for all WorldBank development
+    indicator codes provided in WorldBankIndicators.csv for all years and for
+    all countries provided in WorldBankCountries.csv """
+
+from absl import app
+import pandas as pd
+import itertools
+import requests
+import zipfile
+import io
+import re
+
+# Remaps the columns provided by WorldBank API.
+WORLDBANK_COL_REMAP = {
+    'Country Name': 'CountryName',
+    'Country Code': 'CountryCode',
+    'Indicator Name': 'IndicatorName',
+    'Indicator Code': 'IndicatorCode'
+}
+
+TEMPLATE_TMCF = """
+Node: E:WorldBank->E0
+typeOf: dcs:StatVarObservation
+variableMeasured: C:WorldBank->StatisticalVariable
+observationDate: C:WorldBank->Year
+observationPeriod: "P1Y"
+observationAbout: C:WorldBank->IsoCode
+value: C:WorldBank->Value
+"""
+
+TEMPLATE_STAT_VAR = """
+Node: dcid:WorldBank/{INDICATOR}
+name: "{NAME}"
+description: "{DESCRIPTION}"
+typeOf: dcs:StatisticalVariable
+populationType: dcs:{populationType}
+statType: dcs:{statType}
+measuredProperty: dcs:{measuredProperty}
+measurementDenominator: dcs:{measurementDenominator}
+{CONSTRAINTS}
+"""
+
+COUNTRY_MCF = """
+Node: E:WorldBank->E1
+typeOf: dcs:Country
+countryAlpha3Code: C:WorldBank->IsoCode
+"""
+
+def read_worldbank(country_iso):
+    """ Fetches and tidies all ~1500 Worldbank indicators
+        for a given 3 digit ISO code.
+
+        For a particular 3 digit ISO code, this function fetches the entire ZIP
+        file for that particular country for all Worldbank indicators.
+        The format of the file has rows for each economic indicator for
+        each year. The data is tidied so that each year is a column.
+
+        Args:
+            country_iso: 3-digit ISO code for a country, as a string.
+
+        Returns:
+            A tidied pandas dataframe with all indicator codes for a particular
+            country in the format of (country, indicator, year, value).
+
+        Notes:
+            Takes approximately 10 seconds to download and
+            tidy one country in Google colab.
+    """
+    country_zip = ("http://api.worldbank.org/v2/en/country/"
+        + country_iso + "?downloadformat=csv")
+    r = requests.get(country_zip)
+    filebytes = io.BytesIO(r.content)
+    myzipfile = zipfile.ZipFile(filebytes)
+
+    # We need to select the data file which starts with "API",
+    # but does not have an otherwise regular structure.
+    file_to_open = None
+    for file in myzipfile.namelist():
+        if file.startswith("API"):
+            file_to_open = file
+            break
+    assert file_to_open is not None
```

Comment:
```suggestion
    assert file_to_open is not None, f"Failed to find a file for {country_iso}"
```

I may not have the syntax correct.

Reply:
Added.